### PR TITLE
New feature: Unioning source data

### PR DESCRIPTION
**Pull request by pawelngei**
**Are you a current Fivetran customer?** 
Yes, Pawel Ngei, Pattern Brands

**What change(s) does this PR introduce?** 
- Introduce the ability to union data from multiple schemas databases, as per https://github.com/fivetran/dbt_klaviyo_source/pull/9 (for Klaviyo)

**Does this PR introduce a breaking change?**
- [ ] Yes (please provide breaking change details below.)
- [x] No  (please provide explanation how the change is non breaking below.)

**Is this PR in response to a previously created Issue**
- [ ] Yes, Issue [link issue number here]
- [x] No 

**How did you test the PR changes?** 
- [ ] CircleCi  
- [x] Other (please provide additional testing details below)

Pattern Brands data, two datasets from Facebook Ads connector

**Select which warehouse(s) were used to test the PR**

- [ ] BigQuery
- [ ] Redshift
- [x] Snowflake
- [ ] Postgres
- [ ] Databricks
- [ ] Other (provide details below)

:building_construction: 

**Feedback**

Will need to remember to add a Readme note about the `basic_ad` table to the other two plugins, it's not advertised enough

### CODE REVIEW

File: integration_tests/dbt_project.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: 'facebook_ads_source_integration_tests'
-version: '0.2.3'
```
Comment (pawelngei) [inline integration_tests/dbt_project.yml:2]: That was outdated

File: models/tmp/stg_facebook_ads__account_history_tmp.sql [hunk 2]
```diff
@@ -1 +1,10 @@
-select * from {{ var('account_history') }}
+{{
+    fivetran_utils.union_data(
+        table_identifier='account_history',
+        database_variable='facebook_ads_database',
+        schema_variable='facebook_ads_schema',
+        default_database=target.database,
+        default_schema='facebook_ads',
+        default_variable='account_history'
```
Comment (fivetran-joemarkiewicz) [inline models/tmp/stg_facebook_ads__account_history_tmp.sql:8]: When building out the social media roll up package we found a way to integrate a custom variable for the union_schema and union_databases if this feature is used across packages in the same dbt_project. Would you be able to add the `union_schema_variable` and `union_database_variable` for these `_tmp` models.

[Here](https://github.com/fivetran/dbt_facebook_pages_source/blob/master/models/tmp/stg_facebook_pages__daily_page_metrics_total_tmp.sql#L1-L12) is an example of how we do this within the dbt_facebook_pages_source package. I would envision the variable names to be `facebook_ads_union_schemas` and `facebook_ads_union_databases` respectively.

Let me know if you have any questions on this!

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@pawelngei thanks again so much for opening this PR! We have added a feature to the union_data macro that allows us to define a custom variable for the various sources. It will require adding two lines to the `_tmp` models. Below is an example of the two additional lines:

```sql
        union_schema_variable='facebook_ads_union_schemas',
        union_database_variable='facebook_ads_union_databases'
```

Let me know if you have any questions on this request!

File: models/tmp/stg_facebook_ads__account_history_tmp.sql (hunk 2 above)
Comment (pawelngei) [reply models/tmp/stg_facebook_ads__account_history_tmp.sql:8]: Thank you! I don't have access to my dev machine this week, is it okay if I add this next Monday?

Comment (fivetran-joemarkiewicz) [reply models/tmp/stg_facebook_ads__account_history_tmp.sql:8]: Of course that is perfect. Thanks again!!

Comment (pawelngei) [reply models/tmp/stg_facebook_ads__account_history_tmp.sql]: Implemented, tested locally (without the social media roll-up)!

**Comment by fivetran-joemarkiewicz** [comment]
Thanks again so much for putting the effort into this PR @pawelngei! There are a few minor changes (tests and formatting updates) that I am going to apply before we include this in a live version of the package. Therefore I am changing the base branch to be a working branch that I will apply these updates to.

Thanks so much!

**Pull request merged**